pkgs/applications/misc/obinskit/default.nix

```diff
@@ -0,0 +1,81 @@
+{ lib
+, stdenv
+, fetchzip
+, patchelf
+, xlibs
+, glibc
+, gtk3-x11
+, alsaLib
+, nss
+, libpulseaudio
+, libxkbcommon
+, at_spi2_atk
+, makeWrapper
+, atomEnv
+}:
+
+let
+  libPath = lib.makeLibraryPath [
+    glibc
+    gtk3-x11
+    xlibs.libXtst
+    xlibs.libXt
+    xlibs.libXScrnSaver
+    alsaLib
+    nss
+    libpulseaudio
+    libxkbcommon
+    at_spi2_atk
+  ];
+
+in stdenv.mkDerivation rec {
+  pname = "obinskit";
+  version = "1.1.1";
+
+  src = fetchzip {
+    url = "http://releases.obins.net/occ/linux/tar/ObinsKit_${version}_x64.tar.gz";
+    sha256 = "0wxsrpv2s917zhx8q5r3p136mj1sbm0adby2vqvi2n6af4bqb9aa";
+  };
+
+  buildInputs = [
+    makeWrapper
+  ];
+
+  installPhase = ''
+    mkdir -p $out/lib
+    install $src/libffmpeg.so $out/lib/
+    install $src/libnode.so $out/lib/
+
+    mkdir -p $out/opt/obinskit
+    install $src/obinskit $out/opt/obinskit/
+    install $src/icudtl.dat $out/opt/obinskit/
+    install $src/natives_blob.bin $out/opt/obinskit/
+		install $src/v8_context_snapshot.bin $out/opt/obinskit/
+    install $src/blink_image_resources_200_percent.pak $out/opt/obinskit/
+    install $src/content_resources_200_percent.pak $out/opt/obinskit/
+    install $src/content_shell.pak $out/opt/obinskit/
+    install $src/ui_resources_200_percent.pak $out/opt/obinskit/
+    install $src/views_resources_200_percent.pak $out/opt/obinskit/
+    cp -r $src/resources $out/opt/obinskit/
+    cp -r $src/locales $out/opt/obinskit/
+
+    mkdir -p $out/bin
+    ln -s $out/opt/obinskit/obinskit $out/bin
+
+    patchelf \
+      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      --set-rpath "$out/lib:${atomEnv.libPath}:${libPath}" \
+      $out/opt/obinskit/obinskit
+
+    wrapProgram $out/opt/obinskit/obinskit \
+      --prefix LD_LIBRARY_PATH : "${atomEnv.libPath}:${libPath}"
+  '';
```

Comment:
this is a little redundant with the patchelf command you have, you should be able to remove this

Reply:
this is becuase it's done as part of the install phase, do it as part of the postFixup, or set `dontPatchElf = true;` to solve this

Reply:
the fixUpPhase likes to mess with rpaths, IIRC

Reply:
I've just tried  `patchelf`ing to the `postFixup` phase, and `dontPatchELF = true;`, including both, and no luck. It immediately crashes with:

```
Error: libxkbcommon-x11.so.0: cannot open shared object file: No such file or directory
```

That said, I've remembered that I need to use `strace -f` to get all the processes' outputs, and with that, I've found that I only need to add four libraries to `LD_LIBRARY_PATH` that aren't in `DT_NEEDED`:

```fish
    wrapProgram $out/opt/obinskit/obinskit \
      --prefix LD_LIBRARY_PATH : "${libxkbcommon}/lib:${xlibs.libXt}/lib:${libudev.lib}/lib:${gcc-unwrapped.lib}/lib"
```

And more good news, I've made it run with the `electron_3` Nixpkg binary, which greatly reduces package imports and patching.

Reply:
hmm, you're right :(

cc @worldofpeace You're better at wrapping than I am